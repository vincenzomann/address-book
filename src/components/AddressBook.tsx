import React from 'react';
import { RouteComponentProps } from '@reach/router';
import { useContextProvider } from './../context/Context';
import './sass/AddressBook.scss';
import AddressInfo from './AddressInfo';

interface Props extends RouteComponentProps { }

const AddressBook: React.FC<Props> = () => {

	// Use state from context
	const { addresses, selected, setSelected } = useContextProvider();

	return (
		<div id='addressBook'>
			<h1 className='pageTitle'>
				Address Book
			</h1>
			{/* When user clicks on an address card it is displayed */}
			{selected.id &&
				<div className='selectedAddress'>
					Selected Address:
					<AddressInfo address={selected} />
				</div>
			}
			<div id='addresses'>
				{/* If there are address in saved they are displayed, else shows message */}
				{addresses.length ?
					addresses.map((address) => (
						// If the card id matches the selected id, add styling to the card border
						<div className={`card address ${(address.id === selected.id) ? 'selected' : ''} `}
							key={address.id}
							onClick={() => setSelected(address)}
						>
							<AddressInfo address={address} />
						</div>
					)) : (
						<div>Go to the 'Add Address' page to fill your address book</div>
					)}
			</div>
		</div>
	);
};

export default AddressBook;
