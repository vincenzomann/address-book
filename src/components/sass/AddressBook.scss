#addressBook {
	@media only screen and (min-width: 700px) {
		padding: 1rem;
	}
	#addresses {
		margin: 1rem 0;
		.address {
			cursor: pointer;
			padding: 1rem;
			margin: 1rem 0;
		}
		.selected {
			border-color: #009bc1;
		}
	}
}
