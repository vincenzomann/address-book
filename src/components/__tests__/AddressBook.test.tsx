import React from 'react';
import { shallow } from 'enzyme';
import AddressBook from '../AddressBook';

test('renders the component', () => {
	const component = shallow(<AddressBook />);
	expect(component).toMatchSnapshot();
});
