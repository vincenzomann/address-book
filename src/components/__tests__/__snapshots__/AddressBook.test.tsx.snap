// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  id="addressBook"
>
  <h1
    className="pageTitle"
  >
    Address Book
  </h1>
  <div
    id="addresses"
  >
    <div>
      Go to the 'Add Address' page to fill your address book
    </div>
  </div>
</div>
`;
